internal/job/OrdersStatusJob.go

```diff
@@ -0,0 +1,68 @@
+package job
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/andrey67895/go_diplom_first/internal/config"
+	"github.com/andrey67895/go_diplom_first/internal/database"
+	"github.com/andrey67895/go_diplom_first/internal/helpers"
+	"github.com/andrey67895/go_diplom_first/internal/model"
+)
+
+var client http.Client
+
+func GetAndUpdateOrderStatusByAccrual(login string, number string) (*http.Response, error) {
+	url := config.AccrualSystemAddress + "/api/orders/" + number
+	r, _ := http.NewRequest(http.MethodGet, url, nil)
+	body, err := client.Do(r)
+	if err != nil {
+		helpers.TLog.Error(err.Error())
+	}
+	defer body.Body.Close()
+	if body.StatusCode == http.StatusOK {
+		var tModel model.OrdersAccrualModel
+		err = json.NewDecoder(body.Body).Decode(&tModel)
+		if err != nil {
+			helpers.TLog.Error(err.Error())
+			return body, err
+		}
+		if *tModel.Status == "PROCESSING" || *tModel.Status == "INVALID" || *tModel.Status == "PROCESSED" {
```

Comment:
```suggestion
		if tModel.Status == "PROCESSING" || tModel.Status == "INVALID" || tModel.Status == "PROCESSED" {
```
нет необходимости в явном разыменовании, Go делает это автоматически.

Reply:
Не совсем понял, здесь же подтягиваем данные поля Status, вроде go сам это не делает

`type OrdersAccrualModel struct {
	OrderID *string  `json:"order"`
	Accrual *float64 `json:"accrual"`
	Status  *string  `json:"status"`
}`

Reply:
Пардон, глаз замылился. Все верно, разыменовывание сделано верно, но в таком случае всегда стоит перестраховываться и сначала проверять `tModel.Status != nil && (все остальное)` , иначе можно нарваться на панику.